src/main/java/com/adroll/cantor/HLLCounter.java

```diff
@@ -490,41 +453,48 @@ public static long intersect(List<HLLCounter> hs) {
       return 0;
     }
     for (HLLCounter hll : hs) {
-      if (hll.size() == 0) {
+      if ((hll.getMinHash() == null && hll.size() == 0) || hll.getMinHash().size() == 0) {
```

Comment:
hll.size() estimates the total number of values that this HLL is suppose to be tracking. It's an expensive calculation if all we care about is whether or not it's 0.